source/math/FloatMatrix.ooc

```diff
@@ -51,8 +51,13 @@ FloatMatrix : cover {
 	// <param name="x">Column number of a matrix.</param>
 	// <param name="y">Row number of a matrix.</param>
 	// <returns></returns>
-	get: func@ (x, y: Int) -> Float { this[x, y] } //TODO Deprecated, remove when no longer used
-	operator [] (x, y: Int) -> Float { this elements[x + y * this width] }
+	operator [] (x, y: Int) -> Float {
+		version (safe) {
+			if (x + y * this width >= this width * this height || x + y * this width < 0)
```

Comment:
This doesn't cover the case where both `x` and `y` are negative or only one of `x` or `y` exceed their respective bounds.